src/Pages/BucketListPage.js

```diff
@@ -1,11 +1,74 @@
+import React from "react";
 import { NavLink } from "react-router-dom";
-import { useContext } from "react";
+import { useContext, useState, useEffect } from "react";
 import { UserContext } from "../App.js";
+import BucketForm from "../Components/BucketForm.js";
+import { database } from "../firebase/firebase";
+import { onChildAdded, ref, update } from "firebase/database";
+
+const REALTIME_DATABASE_KEY_BUCKET = "bucket-list";
 
 export default function BucketListPage() {
   //Pull in context from App.js
   const context = useContext(UserContext);
 
+  //create state to view bucket list
+  const [bucketList, setBucketList] = useState([]);
+  const [showBucketForm, setShowBucketForm] = useState(false);
+
+  //to view bucket list
+  useEffect(() => {
+    const bucketListRef = ref(database, REALTIME_DATABASE_KEY_BUCKET);
+
+    onChildAdded(bucketListRef, (data) => {
+      setBucketList((state) => [...state, { key: data.key, val: data.val() }]);
+    });
+  }, []);
+
+  //create a toggle for checkbox
+  const toggleCheckBox = (bucketItemKey, itemId) => {
+    const updatedBucketList = bucketList.map((bucketItem) => {
+      if (bucketItem.key === bucketItemKey) {
+        const updatedItems = bucketItem.val.items.map((item) => {
+          if (item.id === itemId) {
+            return { ...item, completed: !item.completed };
+          }
+          return item;
+        });
+
+        return {
+          ...bucketItem,
+          val: {
+            ...bucketItem.val,
+            items: updatedItems,
+          },
+        };
+      }
+      return bucketItem;
+    });
+
+    // Update the state and Firebase with the updated data
+    setBucketList(updatedBucketList);
+
+    // Update Firebase with the latest state data
+    const updatedData = {
+      [bucketItemKey]: {
+        ...updatedBucketList.find((item) => item.key === bucketItemKey).val,
+      },
+    };
+    update(ref(database, REALTIME_DATABASE_KEY_BUCKET), updatedData);
+  };
+
+  // Function to show the BucketForm
+  const showBucketFormOnClick = () => {
+    setShowBucketForm(true);
+  };
+
+  // Function to hide the BucketForm
+  const hideBucketForm = () => {
```

Comment:
Can be merged with the above code as toggleBucketForm setShowBucketForm(!showBucketForm) so it toggles open/close